Source/WebKit/UIProcess/Automation/BidiBrowsingContextAgent.cpp

```diff
@@ -291,15 +292,29 @@ void BidiBrowsingContextAgent::navigate(const BrowsingContext& browsingContext,
     RefPtr session = m_session.get();
     ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!session, InternalError);
 
-    auto pageLoadStrategy = pageLoadStrategyFromReadinessState(optionalReadinessState.value_or(defaultReadinessState));
-    session->navigateBrowsingContext(browsingContext, url, pageLoadStrategy, defaultPageLoadTimeout.milliseconds(), [url, callback = WTFMove(callback)](CommandResult<void>&& result) {
+    RefPtr webPageProxy = session->webPageProxyForHandle(browsingContext);
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!webPageProxy, WindowNotFound);
+
+    auto waitCondition = optionalReadinessState.value_or(defaultReadinessState);
+    if (optionalReadinessState && *optionalReadinessState == ReadinessState::None)
+        waitCondition = ReadinessState::None;
+    else if (!optionalReadinessState)
+        waitCondition = ReadinessState::Interactive;
+
+    URL baseURL = URL(webPageProxy->currentURL());
+    URL urlRecord(baseURL, url);
+
+    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!urlRecord.isValid(), InvalidParameter);
+
+    auto pageLoadStrategy = pageLoadStrategyFromReadinessState(waitCondition);
```

Comment:
Doesn't the `value_or` already achieve what you want from the `if` and `else if`?

```suggestion
    RefPtr webPageProxy = session->webPageProxyForHandle(browsingContext);
    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!webPageProxy, WindowNotFound);

    URL baseURL(webPageProxy->currentURL());
    URL urlRecord(baseURL, url);
    ASYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!urlRecord.isValid(), InvalidParameter);

    auto pageLoadStrategy = pageLoadStrategyFromReadinessState(optionalReadinessState.value_or(defaultReadinessState));
```

Reply:
Yes you are right, I think when I merged my branches I left this if statement !